### PR TITLE
Edits to nucleation benchmark as proposed by Mancias et al 2022

### PR DESCRIPTION
**Pull request by joseam2**
## Motivation
These edits to the nucleation benchmark problem stem from the new journal publication and presentation given at CHIMAD XIV in October of 2022 which can both be accessed below:

### Paper:
https://doi.org/10.1007/s40192-022-00284-1
Mancias, J., Attari, V., Arróyave, R. et al. On the Effect of Nucleation Undercooling on Phase Transformation Kinetics. Integr Mater Manuf Innov (2022). 

### Presentation:
https://speakerdeck.com/joseam2/potential-edits-to-the-nucleation-benchmark

### Github Repository
Contains code and data from the paper
https://github.com/joseam2/PF_Nucleation_Benchmark_Julia

## Summary of Edits
The edits included are briefly explained below:
- Use periodic boundary conditions for parts 2 and 3
- Change critical radius ( $r ^ *$ ) in part 3 to $r ^ * = 1.0$
- Calculate the Avrami slope using a fitting range in the y-axis ( $\log[-\log(1-Y)]$ ) in parts 2 and 3, the range $-2 < \log[-\log(1-Y)] < 0$
- Increase the amount of simulations to increase statistical accuracy in parts 2 and 3 from $5$ to $10$.

The edits in the file caused by the machine-specific python code are not necessary to be accepted (e.g. execution_count, image.png data, metadata). Therefore, all edits following line 1000 in the jupyter notebook are not necessary.

### CODE REVIEW

**Comment by tkphd** [comment]
Hi @joseam2, thank you for the contribution! Looking very much forward to seeing this deployed.

For readability, can you visit the [GitHub-Flavored Markdown docs](https://docs.github.com/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) and edit the PR description so it renders more properly? Specifically, please insert a space after each hyphen intended as a bullet/list item, and wrap math expressions in `$` so they render as beautifully as possible. Thanks!

**Comment by joseam2** [comment]
> 

I have edited the PR description above so that it renders better! Sorry about that.

File: benchmarks/benchmark8.ipynb [hunk 1]
```diff
@@ -393,14 +396,14 @@
     "## Part (c): multiple seeds at random times\n",
     "\n",
     "For the third part of the homogeneous benchmark problem, the computational domain is first expanded to $1000 \\times1000$. Instead of inserting nuclei at fixed time $t = 0$ as in part two of the problem, in this case 100 random nucleation times $t_i$ are generated, $i=1,\\ldots,100$, drawn from a uniform distribution in the interval $t_i \\in [0, 600)$ with centers $\\mathbf{r}_i$ drawn from a uniform distribution on the $1000\\times1000$ domain.\n",
-    "We decrease the driving force to $\\Delta f=1/(6\\sqrt{2})$, which corresponds to a critical radius or $r^*=2$, but keep the initial radii of the inserted nuclei the same as those in the second part of the problem\n",
+    "The driving force is the same as Part (b): $\\Delta f=1/(3 \\sqrt{2})$, which corresponds to a critical radius or $r^*=1$. The initial radii are also kept the same at $r_0 = 2.2r^*$ and the boundary conditions are periodic. \n",
```
Comment (tkphd) [inline benchmarks/benchmark8.ipynb:399]: While we're amending this sentence, let's drop a quirk of grammar
```suggestion
    "The driving force is the same as Part (b): $\\Delta f=1/(3 \\sqrt{2})$, which corresponds to a critical radius $r^*=1$. The initial radii are also kept the same at $r_0 = 2.2r^*$ and the boundary conditions are periodic. \n",
```

File: benchmarks/benchmark8.ipynb [hunk 2]
```diff
@@ -393,14 +396,14 @@
...
     "\\begin{equation}\n",
     "    \\phi_i(r)=\\frac{1}{2}\\left[1-\\tanh\\left(\n",
     "    \\frac{|\\mathbf{r}-\\mathbf{r}_i|-1.1r^*}{\\sqrt{2}}\n",
     "    \\right)\\right].\n",
     "\\end{equation}\n",
     "\n",
-    "Again, $\\phi$ is set to unity in regions of overlaps of nuclei. The simulations are run up to times $t=600$, and repeated for four more random initial conditions.  The total free energy plot the total free energy, the time evolution of the solid fraction $Y$ and the discrete particle count $N$, are plotted as functions of time, and an  Avrami plot is generated for the five runs.\n",
+    "Again, $\\phi$ is set to unity in regions of overlaps of nuclei. The simulations are run up to times $t=600$, and repeated for four more random initial conditions.  The total free energy plot the total free energy, the time evolution of the solid fraction $Y$ and the discrete particle count $N$, are plotted as functions of time, and an  Avrami plot is generated for each of the $P$ runs. Once again, $P = 10$ is recommended.\n",
```
Comment (tkphd) [inline benchmarks/benchmark8.ipynb:406]: "four more" is based on $P=5$. Fix copy-paste error in the original.

```suggestion
    "Again, $\\phi$ is set to unity in regions of overlaps of nuclei. The simulation is performed up to times $t=600$, and then repeated for a total of ten (10) simulations for statistical significance. The total free energy, time evolution of the solid fraction $Y$, and discrete particle count $N$ are plotted as functions of time, and an Avrami plot is generated for each of the 10 runs.\n",
```

**Review by tkphd — CHANGES_REQUESTED** [review]
A few tweaks, but this looks great!

I've dropped the $P$ nomenclature, since the specification uses a lot of variables already and it sows uncertainty over "how many am I really supposed to do?"
Finite differencing codes may not be able to finish 10 iterations of Part (c) in any reasonable amount of time -- which is a valuable point to consider when choosing a code.

**Comment by joseam2** [comment]
That is a great point about the $P$ nomenclature, thank you. I have accepted all of your requested changes in github.

**Comment by wd15** [comment]
@tkphd, @joseam2

[Link to the notebook rendered as HTML](https://random-cat-1465.surge.sh/benchmarks/benchmark8.ipynb/)

**Review by wd15 — APPROVED** [review]

**Comment by joseam2** [comment]
@wd15 @tkphd Damien pointed out to me that there were issues in the Submission Guidelines section which reflected the original value of 5 simulations. Those values have been replaced with 10 simulations in each instance in that section. Sorry about that, my latest commit was made to rectify this.

**Pull request merged**